pkg/endpoint/policy.go

```diff
@@ -793,10 +795,10 @@ func (e *Endpoint) Regenerate(owner Owner, reason string) <-chan bool {
 	go func(owner Owner, req *Request, e *Endpoint) {
 		buildSuccess := true
 
-		e.Mutex.Lock()
+		e.Mutex.RLock()
 		// This must be accessed in a locked section, so we grab it here.
 		scopedLog := e.getLogger()
-		e.Mutex.Unlock()
+		e.Mutex.RUnlock()
```

Comment:
This locking is no longer needed as `e.getLogger()` takes care of it's own locking.